src/chocolatey/infrastructure.app/builders/ConfigurationBuilder.cs

```diff
@@ -457,6 +460,21 @@ private static void SetGlobalOptions(IList<string> args, ChocolateyConfiguration
                 });
         }
 
+        private static void SetProxyOptions(ChocolateyConfiguration config, Container container)
+        {
+            // Evaluation order of Proxy settings: System Set -> Environment Variable Set -> Chocolatey Configuration File Set -> CLI Passed in Argument
+            // If we don't yet have a Proxy Location, check if the system has one configured in the registry
+            if (string.IsNullOrWhiteSpace(config.Proxy.Location))
+            {
+                var registryService = container.GetInstance<IRegistryService>();

```

Comment:
This should be gated behind an OS type check. The registry service is not available on Linux/Mac

Reply:
~~Well that sounds like a Linux/Mac problem~~ Good catch. I've added an OS check to this if block.

Reply:
That is a great catch @TheCakeIsNaOH !